PR: https://github.com/odoo/odoo/pull/225669

From: 136118c71a0142302c5f9bb97606e80e9cc9eb7a
From: Lulu Grimalkin (lugr)
Date: 2026-02-05 12:06:06

Breaking data model changes scores: del:4 + add:0, change matches:
-    finished_lot_ids = fields.Many2many(
-    order_finished_lot_ids = fields.Many2many('stock.lot', string="Finished Lot/Serial Number", related="raw_material_production_id.lot_producing_ids")

Total Changes: 12

[REM] mrp{,_subcontracting}: remove `{order_,}finished_lot_ids` fields

The (related) fields `stock_move.order_finished_lot_ids` and
`mrp_workorder.finished_lot_ids` are being removed for being superfluous
after the workorder tablet views were scrapped.

Task ID: 5025510

closes odoo/odoo#225669

Related: odoo/enterprise#93977
Related: odoo/upgrade#8387
Signed-off-by: Tiffany Chang (tic) <tic@odoo.com>

================================= pseudo patch: =================================

--- a/addons/mrp/models/mrp_workorder.py
+++ b/addons/mrp/models/mrp_workorder.py
@@ -111,10 +111,6 @@ class MrpWorkorder(models.Model):
     move_line_ids = fields.One2many(
         'stock.move.line', 'workorder_id', 'Moves to Track',
         help="Inventory moves for which you must scan a lot number at this work order")
-    finished_lot_ids = fields.Many2many(
-        'stock.lot', string='Lot/Serial Numbers', related='production_id.lot_producing_ids',
-        domain="[('product_id', '=', product_id), ('company_id', '=', company_id)]",
-        readonly=False, check_company=True)
     time_ids = fields.One2many(
         'mrp.workcenter.productivity', 'workorder_id', copy=False)
     is_user_working = fields.Boolean(
@@ -464,13 +460,6 @@ class MrpWorkorder(models.Model):
         )
         return interval['hours'] * 60
 
-    @api.onchange('finished_lot_ids')
-    def _onchange_finished_lot_ids(self):
-        if self.production_id:
-            res = self.production_id._can_produce_serial_numbers(sns=self.finished_lot_ids)
-            if res is not True:
-                return res
-
     def write(self, vals):
         values = vals
         new_workcenter = False

--- a/addons/mrp/models/stock_move.py
+++ b/addons/mrp/models/stock_move.py
@@ -51,7 +51,6 @@ class StockMove(models.Model):
         'mrp.bom.byproduct', 'By-products', check_company=True,
         help="By-product line that generated the move in a manufacturing order")
     unit_factor = fields.Float('Unit Factor', compute='_compute_unit_factor', store=True)
-    order_finished_lot_ids = fields.Many2many('stock.lot', string="Finished Lot/Serial Number", related="raw_material_production_id.lot_producing_ids")
     should_consume_qty = fields.Float('Quantity To Consume', compute='_compute_should_consume_qty', digits='Product Unit')
     cost_share = fields.Float(
         "Cost Share (%)", digits=(5, 2),  # decimal = 2 is important for rounding calculations!!
